src/lib/services/invoice-service.js

```diff
@@ -200,21 +202,20 @@ const getInvoicesForBatchDownload = async batch => {
  * @param {Object} options
  * @param {Boolean} options.includeTransactions
  * @param {Boolean} options.includeInvoiceAccounts - whether to load invoice data from CRM
- * @return {Promise<Invoice>}
+ * @return {Array<Invoice>}
  */
 const getInvoicesForBatch = async (batch, options = {}) => {
   const defaults = {
     includeTransactions: false,
     includeInvoiceAccounts: false
   };
 
-  const context = { batch, options: Object.assign(defaults, options) };
+  let context = { batch, options: Object.assign(defaults, options) };
 
-  return pWaterfall([
-    getBatchInvoices,
-    getCRMData,
-    mapToInvoices
-  ], context);
+  context = await getBatchInvoices(context);
```

Comment:
This is the start of an effort to actively remove: 'p-waterfall'

My understanding may be wrong but is is pointless and adds another layer of complexity where it is not needed ?

Reply:
Kept the context object for now. But i would look to isolate the  return objects from function calls in the future.